.github/workflows/compatibility-report.yml

```diff
@@ -0,0 +1,64 @@
+name: Generate Compatibility Report
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo_target:
+        description: Generate the report for the local repo, the docs repo, or both.
+        required: true
+        type: choice
+        default: both
+        options:
+          - local
+          - docs
+          - both
+
+  push:
+    branches:
+      - main
+    paths:
+      - 'test/versioned/**/package.json'
```

Comment:
This is saying: "every time a push to the `main` branch occurs (i.e. a pull request was merged into `main`), and there have been changes to any `package.json` file that resides under `test/versioned/`, run the workflow".

We can add a scheduled run of this if it is desired. It would be a separate "on" block like the `workflow_dispatch` block.